sympy/printing/pycode.py

```diff
@@ -504,6 +504,37 @@ def _print_MatrixBase(self, expr):
             func = self._module_format('numpy.array')
         return "%s(%s)" % (func, self._print(expr.tolist()))
 
+    def _print_CodegenArrayTensorProduct(self, expr):
+        array_list = [j for i, arg in enumerate(expr.args) for j in
+                (self._print(arg), "[%i]" % i)]
+        return "%s(%s)" % (self._module_format('numpy.einsum'), ", ".join(array_list))
```

Comment:
I'm trying to figure out how this works. Shouldn't einsum take a subscript string describing the contraction as the first argument?

Reply:
There are two formats for `einsum`: with the string or with the axes numbers after each array. I chose the second one to avoid the limitation of 26 axes.

Reply:
I see now. That's pretty well hidden in the einsum docstring.

Reply:
Yes, I didn't notice it at first. `einsum` appears to have a bug at handling Python objects, maybe the code generator should rather focus on `numpy.tensordot`.

Reply:
What is the difference? Based on the docstrings, it seems einsum has the ability to do optimizations and tensordot does not. Also what is the bug in NumPy?